joomla/joomla-cms
# [3.10] Joomla.Showon.initialise API, and  make sure that showon initialised only once, fork of 18843

Pull Request for Issue #18843 and only for J3 .

### Summary of Changes

This is fork of #18843, with requested changes.
The code make sure that showon initialised only once, no mater how often Joomla.setUpShowon was called.
This is achieved with extra flag `showonInitialised` that sets while showon initialisation.


### Testing Instructions

Make sure `showon` still works.

Example, go to global config change "Debug Language" on/off, should downup/hides "Language Display" field.
Also the field "Language Display"  will have extra attribute `data-showon-initialised`


### Actual result BEFORE applying this Pull Request
showon works,
the showon field do not have `data-showon-initialised`



### Expected result AFTER applying this Pull Request
showon works,
the showon field do have `data-showon-initialised` attribute
